static/js/controllers/contacts-content.js

```diff
@@ -147,24 +147,17 @@ var _ = require('underscore'),
         return fields;
       };
 
-      var shiftAndMarkContactPerson = function(children, parent) {
-        if (!children || !children.length) {
-          return;
-        }
-        var contactPersonId = parent.contact && parent.contact._id;
-        if (!contactPersonId) {
-          return;
+      var getPrimaryContact = function(doc) {
+        if (!doc || !doc.contact || !doc.contact._id) {
+          return $q.resolve();
         }
-        var primaryContactIdx = _.findIndex(children, function(child) {
-          return child.doc._id === contactPersonId;
-        });
-        if (primaryContactIdx < 0) {
-          return;
-        }
-        var primaryContact = children[primaryContactIdx];
-        primaryContact.isPrimaryContact = true;
-        children.splice(primaryContactIdx, 1);
-        children.unshift(primaryContact);
+        // Fetch the contact person from db instead of using the contents of the contact field.
+        // Because in 2017 we stop copying stuff in docs.
+        return DB().get(doc.contact._id)
+          .catch(function(err) {
+            $log.error(err);
+            return;
```

Comment:
NB: Don't think you need this return.

Also, are we sure we want to swallow the error here? That means that to the caller of this function, there is no difference between "you have no doc / contact" and "there was an error when I hit the DB"